File: app/src/main/java/com/harvey/arcface/utils/MainHandler.java
package com.harvey.arcface.utils;

import android.os.Handler;
import android.os.Looper;

public class MainHandler extends Handler {
	private static MainHandler instance = new MainHandler(Looper.getMainLooper());

	protected MainHandler(Looper looper) {
		super(looper);
	}

	public static MainHandler getInstance() {
		return instance;
	}

	public static void run(Runnable runnable) {
		if (Looper.getMainLooper().equals(Looper.myLooper())) {
			runnable.run();
		} else {
			instance.post(runnable);
		}

	}
}

File: kotlin-library/src/main/java/com/harvey/arcface/MainHandler.kt
package com.harvey.arcface

import android.os.Looper

/**
 * Created by hanhui on 2018/6/1 0001 09:51
 */
class MainHandler private constructor(looper: Looper) : android.os.Handler(looper) {

    companion object {
        
        private val instance = MainHandler(Looper.getMainLooper())

        fun run(r: Runnable) {
            if (Looper.getMainLooper() == Looper.myLooper()) {
                r.run()
            } else {
                instance.post(r)
            }
        }

        fun runDelayed(r: Runnable, delayMillis: Long) {
            instance.postDelayed(r, delayMillis)
        }
    }
}
